lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -97,9 +96,9 @@ public function tryMethod()
     /**
      * Lists the available databases for this connection.
      *
-     * @return string[]
+     * @return array<int, string>
```

Comment:
This is more of a leading question than a suggestion: do we really need to enforce the type of the key if it's not used? If enforced, it may force unnecessary changes to already correct code (I'm referring to the newly introduced `array_values()`).

Reply:
Hmm, good question. I think yes, because without it, if you introduce new code that uses that key, phpstan won't validate that you are using the type of the key correctly.

Reply:
But you don't know the type of the key before you introduce it. And once you do, if the signature says a method accepts `array<int|string, T>`, it will have to be able to handle both `int` and `string` keys correctly.

Reply:
Another reason to not introduce key types just everywhere is that `array<K, T>` implies that the method accepts (or returns) a map (correspondence) between keys and values and the order of the key-value pairs is irrelevant (or is at least secondary). While `T[]` means that it's an ordered list of `T` and the order matters while the keys don't.

Reply:
Hmm, I am not sure. I started doing this because @Ocramius suggested it in another pull request. Without the key type, the user can pass in anything they want. Adding the type made sense because it makes the usage explicit.

Reply:
Which is fine as long as the key is not used (`T[]`). Having to put `array_values()` everywhere IMO indicates a problem.

Reply:
If we can guarantee `array<int, T>`, then it should be like that, as it allows downstream to use it in variadic calls. Effectively, the `int` or `string` key is something that will be a BC break downstream regardless, if we change it, so if we can guarantee it by declaring it in first place, that is a win for us.

Reply:
To make the above comment more explicit: having `array<int|string, T>` and changing implementation in the method to return `array<int, T>` to `array<string, T>` is technically not a BC break, but in practice it is one for downstream. If we can restrict types under our control, we should do so to allow these BC breaks to become explicit (when the docblock changes)